azure-pipelines.yml

```diff
@@ -26,6 +26,13 @@ strategy:
        COMPILER: 'clang++'
```

Comment:
With the new clang 10 tests, I think we can drop the clang 4 ones.

Reply:
Clang4 is dropped in the `std::variant` PR because it doesn't have full support for C++17, so once that PR and this one go through we'll be only using Clang 6 and 10